### PR TITLE
Progress bars: center labels vertically and set height on parent .progress

### PR DESCRIPTION
**Pull request by davethegr8**
These changes address the suggestions outlined in #22702

The changes include:

- Setting labels to center vertically: There is no visual difference in the base case, but in cases where the height of the bar is not the default, behaves better.

Before:

screen shot 2017-05-26 at 10 58 37

After: 

screen shot 2017-05-26 at 10 59 22

- Moving where custom heights are set from `.progress-bar` to `.progress`. Again, there is no change in the base case, but on non-default heights it behaves better (more flexible/ maintainable). This is especially useful when using multiple bars.

Before (you have to set them all to be the same height):

```
<div class="progress">
  <div class="progress-bar" style="width: 15%; height: 3rem;">1</div>
  <div class="progress-bar bg-success" style="width: 30%; height: 3rem;">2</div>
  <div class="progress-bar bg-info" style="width: 20%; height: 3rem;">3</div>
</div>
```

After (only one change):

```
<div class="progress" style="height: 3rem;">
  <div class="progress-bar bg-warning" style="width: 15%;">1</div>
  <div class="progress-bar bg-success" style="width: 30%;">2</div>
  <div class="progress-bar bg-info" style="width: 20%;">3</div>
</div>
```

You can still change the heights to be different if you want, but it's less work in the base case.

```
<div class="progress" style="height: 3rem;">
  <div class="progress-bar bg-success" style="width: 30%;">2</div>
  <div class="progress-bar bg-info" style="width: 20%; height: 1rem;">3</div>
</div>
```

screen shot 2017-05-26 at 11 04 49

### CODE REVIEW

File: scss/_progress.scss [hunk 1]
```diff
@@ -1,32 +1,37 @@
+// Progress animations
```
Comment (mdo) [inline scss/_progress.scss]: You can drop these newly added comments—they don't add any new context that the code itself doesn't already tell you :).

File: scss/_progress.scss [hunk 2]
```diff
@@ -1,32 +1,37 @@
...
-  @include box-shadow($progress-box-shadow);
 }
 
 .progress-bar {
-  height: $progress-height;
-  line-height: $progress-height;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  height: 100%;
```
Comment (mdo) [inline scss/_progress.scss]: This isn't necessary as far as I can tell. See https://jsbin.com/wahusavehu/edit?html,css,output.

File: scss/_progress.scss [hunk 3]
```diff
@@ -1,32 +1,37 @@
...
 }
 
+// Basic progress bar
 .progress {
   display: flex;
+  align-items: center;
+  height: $progress-height;
   overflow: hidden; // force rounded corners by cropping it
   font-size: $progress-font-size;
   line-height: $progress-height;
```
Comment (mdo) [inline scss/_progress.scss]: Probably safe to remove this? Or we might need to change the value at least.

**Review by mdo — COMMENTED** [review]
Thanks for tackling this! Just a couple questions here.

File: scss/_progress.scss (hunk 3 above)
Comment (davethegr8) [inline scss/_progress.scss]: ah, it looks like align-items is indeed safe to remove, but the height is needed (empty bars will collapse to zero) with no explicit height.

File: scss/_progress.scss (hunk 2 above)
Comment (davethegr8) [inline scss/_progress.scss]: Great, I'll remove height. :)

File: scss/_progress.scss (hunk 1 above)
Comment (davethegr8) [inline scss/_progress.scss]: Oh whoops, sorry. I was editing the npm package for testing in my project, and those snuck into the dist scss files.

File: scss/_progress.scss (hunk 3 above)
Comment (mdo) [inline scss/_progress.scss]: Height is good to stay here, but I meant we can probably remove the `line-height`.

Comment (davethegr8) [inline scss/_progress.scss]: oh, gotcha, removing. (align-items seems to _also_ be safe with having it on the `.progress-bar`)

File: scss/_progress.scss [hunk 4]
```diff
@@ -5,21 +5,19 @@
...
 
 .progress-bar {
-  height: $progress-height;
-  line-height: $progress-height;
+  display: flex;
+  align-items: center;
+  justify-content: center;
   color: $progress-bar-color;
   background-color: $progress-bar-bg;
-  @include transition($progress-bar-transition);
```
Comment (tmorehouse) [inline scss/_progress.scss]: Any particular reason why the transition was removed from progress in this PR?

Comment (davethegr8) [reply scss/_progress.scss]: I don't believe so, probably an oversight on my part.

**Pull request merged**